### DISCUSSION TITLE
Error Route API Handlers | App Directory

### DISCUSSION THREAD

**Discussion by smo043** [vercel/next.js#47356] [Help] [answered]
I am using the most recent version of Next.Js and moving pages/api to new Route Handlers. However an error occured.

'common.register' is not a function : .next/server/webpack-runtime.js

The logic works perfectly in pages/api, but not in new route handlers. I'm looking for assistance.

src/app/api/hello/route.ts

import {NextResponse} from 'next/server';
import git from 'git-rev-sync';

export function GET() {
 return NextResponse.json({
  git: {
   branch: git.branch(),          
   shortHash: git.short(),   
   hash: git.long(),
   date: git.date()
  }
 });
}

**Comment by icyJoseph** [comment, marked as answer]
Hi,

I do believe the issue has to do with the Webpack configuration used in the app dir, and CommonJS packages.

Credit where it is due: https://github.com/vercel/next.js/issues/44273

So you need to tweak the `webpack` configuration of your `next.config.js`:

```js
/** @type {import('next').NextConfig} */
const nextConfig = {
  experimental: {
    appDir: true,
  },
  webpack: (config) => {
    config.externals.push({
      "git-rev-sync": "commonjs git-rev-sync",
    });
    return config;
  },
};

module.exports = nextConfig;

```

**Reply by smo043** [reply]
Thank you, I will give a try

**Reply by smo043** [reply]
It works, thank you
